react/src/components/ServiceLauncherPageContent.tsx

```diff
@@ -726,7 +726,7 @@ const ServiceLauncherPageContent: React.FC<ServiceLauncherPageContentProps> = ({
                         name="serviceName"
                         rules={[
                           {
-                            pattern: /^(?=.{4,24}$)\w[\w.-]*\w$/,
+                            pattern: /^(?=.{4,24}$)\w[\w-]*\w$/,
```

Comment:
Let's separate the rule into 3 rules:
- Minimum
- Maximum
- Allowed characters
Without this separation, users will receive a long validation error message when typing less than 4 characters, even if it doesn't have any non-allowed characters.

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/XqC2uNFuj0wg8I60sMUh/d32a6ee5-da7c-4e59-9849-d8b8c0bb9081.png)

Reply:
I have separated one more rule. Start/end with a hyphen.